designs/2019-linter-options/README.md

```diff
@@ -0,0 +1,126 @@
+- Start Date: 2019-05-12
+- RFC PR: (leave this empty, to be filled in later)
+- Authors: Toru Nagashima ([@mysticatea](https://github.com/mysticatea))
+
+# The `linterOptions` property in Config Files
+
+## Summary
+
+This RFC adds `linterOptions` property to config files. People can use config files (including shareable configs!) instead of CLI options in order to configure some linter behavior.
+
+## Motivation
+
+We cannot configure some linter's behavior with config files, especially, shareable configs. It's convenient if we can configure those in shareable configs.
+
+## Detailed Design
+
+This adds `linterOptions` property to config files with three properties.
+
+```js
+// .eslintrc.js
+module.exports = {
+    linterOptions: {
+        allowInlineConfig: true,              // Corresponds to --no-inline-config / `options.allowInlineConfig`
+        reportUnusedDisableDirectives: "off", // Corresponds to --report-unused-disable-directives / `options.reportUnusedDisableDirectives`
+        ignorePatterns: [],                   // Corresponds to --ignore-pattern / `options.ignorePattern`
```

Comment:
If a shareable config sets `ignorePattern: ["foo", "bar"]`, and the end user's config sets `ignorePattern: ["foo"]`, how are they merged? It seems important to be able to override the setting that the shareable config provides.

Reply:
Oh, I should describe it.

ESLint concatenates all ignore patterns from all of `.eslintignore`, `--ignore-path`, `--ignore-pattern`, and `linterOptions.ignorePatterns`. If there are multiple `linterOptions.ignorePatterns`, all of them are concatenated.

If people wanted to disable shareable config's setting, they can use negative patterns.